include/maths/CLocalOutlierFactors.h

```diff
@@ -284,28 +284,32 @@ class MATHS_EXPORT CLocalOutlierFactors {
         }
 
         TPointVec projection(projectedDimension, las::zero(points[0]));
-        for (std::size_t i = 0; i < coordinates.size(); /**/) {
+        for (std::size_t bag = 0, i = 0; bag < bags && i < coordinates.size(); /**/) {
             // Create an orthonormal basis for the bag.
             for (std::size_t p = 0; p < projectedDimension; ++p) {
                 for (std::size_t d = 0; d < dimension; ++i, ++d) {
                     projection[p](d) = coordinates[i];
                 }
             }
-            CGramSchmidt::basis(projection);
+            if (COrthogonaliser::orthonormalBasis(projection) &&
+                projection.size() == projectedDimension) {
```

Comment:
`COrthogonaliser::orthonormalBasis` will resize it if there is linear dependency to working precision; `projection` is passed by non-constant reference. Arguably, this could return the modified projection to make this more explicit, but the code base uses the paradigm of (input/)output parameters in lots of places.

Reply:
ah, protection, I understand. This sort of mutation pattern is always a bit...touchy. Performant, but tricky.